src/code/z_debug_display.c

```diff
@@ -1,3 +1,7 @@
+#include "config.h"
+
+#ifndef NO_DEBUG_DISPLAY
```

Comment:
Same thing here, may be better to just ifdef this entire file in the spec rather than in the body of the file.

Reply:
does ifdeffing in spec "prevent" the compilation of the file? because that's the only reason why I did this to begin with, compiling code you won't use isn't great

Reply:
That’s fair I guess. It would prevent linking it. I just don’t like the idea of linking empty files a whole lot is all but perhaps you’re right. We can keep it how you have it here.

Reply:
tbh I just forgot the spec file existed at all lmao, it's weird to link empty files I agree, I'll add more ifdefs in spec and re-arrange the order like you mentioned, creating a "debug" segment in spec might be worth

Reply:
Well the spec has to be ordered a certain way for some things, but if you want you can reorder where the .o's for debug related things are linked in `code`, if that makes sense.